lib/omnibus/config.rb

```diff
@@ -432,10 +432,16 @@ def reset!
       ['omnibus-software']
     end
 
-    # The solaris compiler to use
+    # Solaris linker mapfile to use, if needed
+    # see http://docs.oracle.com/cd/E23824_01/html/819-0690/chapter5-1.html
+    # Path is relative to the 'files' directory in your omnibus project
+    #
+    # For example:
+    #
+    #     /PATH/files/my_map_file 
     #
     # @return [String, nil]
-    default(:solaris_compiler, nil)
+    default(:solaris_linker_mapfile, nil)
```

Comment:
To follow idioms we used in other config values, let's just set a default value here and then check for it's existence when it's used in `Omnibus::Software#with_standard_compiler_flags `:

```ruby
# Solaris linker mapfile to use during compilation. This is relative {#project_root}.
#
# @see http://docs.oracle.com/cd/E23824_01/html/819-0690/chapter5-1.html
# @return [String]
default(:solaris_linker_mapfile, 'files/mapfile/solaris')
```